plugins/inputs/udp_listener/udp_listener.go

```diff
@@ -111,20 +119,29 @@ func (u *UdpListener) Stop() {
 func (u *UdpListener) udpListen() error {
 	defer u.wg.Done()
 	var err error
+
 	address, _ := net.ResolveUDPAddr("udp", u.ServiceAddress)
 	u.listener, err = net.ListenUDP("udp", address)
+
 	if err != nil {
-		log.Fatalf("ERROR: ListenUDP - %s", err)
+		log.Fatalf("E! Error: ListenUDP - %s", err)
 	}
+
 	log.Println("I! UDP server listening on: ", u.listener.LocalAddr().String())
 
 	buf := make([]byte, UDP_MAX_PACKET_SIZE)
+
 	for {
 		select {
 		case <-u.done:
 			return nil
 		default:
 			u.listener.SetReadDeadline(time.Now().Add(time.Second))
+
+			if u.UDPBufferSize > 0 {
+				u.listener.SetReadBuffer(u.UDPBufferSize) // if we want to move away from OS default
```

Comment:
I think SetReadBuffer can return an error, probably best to check and log it if it errors